ETL/Pipeline/extract.py

```diff
@@ -0,0 +1,61 @@
+"Extract Script: Pulls current price and sale status from ASOS API"
+from datetime import datetime
+import requests
+from lambda_multiprocessing import Pool
+
+
+def get_url(product_id: int) -> str:
+    """Returns the API URL for a given product on the ASOS website."""
+    if not isinstance(product_id, int):
+        raise TypeError('Product id must be a integer to get the url.')
+
+    return f"https://www.asos.com/api/product/catalogue/v4/stockprice?productIds=\
+        {product_id}&store=COM&currency=GBP&keyStoreDataversion=ornjx7v-36&country=GB"
+
+
+def get_product_info(product_data: int, headers: dict) -> int:
+    """Gets the price information for a specified product from the ASOS API."""
+    price_endpoint = get_url(product_data['product_id'])
+
+    response = requests.get(price_endpoint, headers, timeout=60)
+
+    if 'errorCode' in response:
+        raise ValueError(f"API error: {response['errorMessage']}")
+    return response.json()[0]
+
+
+def get_current_price(product_info: dict) -> int:
+    """Extracts the current price of the product from the product information."""
+    try:
+        return product_info["productPrice"]["current"]["value"]
+    except KeyError as e:
+        return f"Error getting current price: {e}"
```

Comment:
Just wondering why using try and except over .get

Reply:
The try block allows the code to attempt to access the nested value. If it succeeds, it returns the value. If it fails it raises a KeyError, the except block catches the error and logging the information and handles so it's just skipped.